controller/utils.go

```diff
@@ -122,104 +122,119 @@ func replacePodsAndWait(
 	goroutinePoolSize int,
 ) (timedout, canceled bool, err error) {
 	logger.Debug("starting to replace pods with new ones")
-	stop := make(chan struct{}, 1)
-	defer close(stop) // Dont leak goroutines
-	finishedReplace := make(chan struct{})
 
-	timedoutChan := make(chan bool)
-	canceledChan := make(chan bool)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	finishedReplace := make(chan struct{})
+	canceledChan := make(chan struct{})
 	errChan := make(chan error)
 
 	pods := make(chan *models.Pod, len(podsChunk))
 	for _, pod := range podsChunk {
 		pods <- pod
 	}
 
+	inRollingUpdate := operationManager != nil
```

Comment:
Are there cases where this function will be called without a rolling update?

Reply:
Yes, when it's a rollback. There's no operation on a rollback.